pandas/core/indexes/api.py

```diff
@@ -222,7 +222,7 @@ def union_indexes(indexes, sort: bool | None = True) -> Index:
     if len(indexes) == 1:
         result = indexes[0]
         if isinstance(result, list):
-            result = Index(sorted(result))
+            result = Index(result)
```

Comment:
Pretty sure that this will break tests. If sort=True we should still sort.

Please link the issue in your pr